include/seqan3/core/algorithm/parameter_pack.hpp

```diff
@@ -0,0 +1,112 @@
+ *
+ * \param  fn               The function to call on every argument.
+ * \param  args             The arguments as parameter pack.
+ *
+ * \details
+ *
+ * This function behaves like std::for_each but on parameter packs. The invocation(s) will be done without any loop.
+ *
+ * \include test/snippet/core/algorithm/for_each_value.cpp
+ *
+ * \attention The order of arguments is different to std::for_each. In std::for_each the first argument is a range and
+ * the second one is a callable. Due to limitations with the ordering for parameter packs we had to change this order in
+ * the other way around.
+ *
+ * \sa https://en.cppreference.com/w/cpp/language/parameter_pack
+ */
+template <typename unary_function_t, typename ...args_t>
+//!\cond
+    requires (std::Invocable<unary_function_t, args_t> && ... && true)
+//!\endcond
+constexpr void for_each_value(unary_function_t && fn, args_t && ...args)
+{
+    [[maybe_unused]] int r = (fn(std::forward<args_t>(args)), ..., 0);
+}
+
+/*!\brief Applies a function to a range of types.
+ * \ingroup algorithm
+ *
+ * \tparam ...types         The parameter pack of types.
+ * \tparam unary_function_t The function type, like function pointers, functors and lambdas.
+ *
+ * \param  fn               The function to call on every std::type_identity<type>.
+ *
+ * \details
+ *
+ * This function behaves like std::for_each but on types. The type will be wrapped into std::type_identity and passed as
+ * argument. The invocation(s) will be done without any loop.
+ *
+ * In contrast to `meta::for_each`, this function can handle types which are incomplete, forward declared and not
+ * std::Semiregular.
+ *
+ * \include test/snippet/core/algorithm/for_each_type.cpp
+ *
+ * \sa seqan3::detail::for_each_value
+ */
+template <typename ...types, typename unary_function_t>
+//!\cond
+    requires (std::Invocable<unary_function_t, std::type_identity<types>> && ... && true)
+//!\endcond
+constexpr void for_each_type(unary_function_t && fn)
+{
+    for_each_value(std::forward<unary_function_t>(fn), std::type_identity<types>{}...);
+}
+
+/*!\brief Applies a function to a range of types contained in a list.
+ * \ingroup algorithm
+ *
+ * \tparam type_list_t      The name of the list type.
+ * \tparam ...types         The parameter pack of types within a list.
+ * \tparam unary_function_t The function type, like function pointers, functors and lambdas.
+ *
+ * \param  fn               The function to call on every std::type_identity<type>.
+ * \param  type_list        The list of types, i.e. seqan3::type_list, std::tuple. [unused]
+ *
+ * \details
+ *
+ * This function behaves like std::for_each but on types. The type will be wrapped into std::type_identity and passed as
+ * argument. The invocation(s) will be done without any loop.
+ *
+ * In contrast to `meta::for_each`, this function can handle types which are incomplete, forward declared and not
```

Comment:
Remove the `meta::for_each` reference, because when we are going to remove the dependency, this comment get's kind of outdated.

Reply:
@smehringer :sob: :cry: